src/types.ts

```diff
@@ -137,6 +162,190 @@ type FinalAuthReqsSatisfied<T extends HasFinalAuthorize<any, any>> = IntersectPr
 */
 // Start with the "optionalized" section below
 
+type ObjectValueToAny<T, TKeyProperty extends string | symbol | number> = [
+  T
+] extends [object]
+  ? { [key in keyof T]: any }
+  : Record<TKeyProperty, T>;
+
+type NotObjectCase<TOut, TKeyProperty extends string | symbol | number> = [
+  TOut
+] extends [object]
+  ? TOut
+  : Record<TKeyProperty, TOut>;
+
+type ReturnedSomewhereUpToPreAuthorizeAuthorize<
+  T,
+  TOut,
+  TKeyProperty extends string | symbol | number
+> = [T] extends [HasSanitizeBody<any, ObjectValueToAny<TOut, TKeyProperty>>]
+  ? HasSanitizeBody<any, NotObjectCase<TOut, TKeyProperty>>
+  : [T] extends [HasSanitizeParams<any, ObjectValueToAny<TOut, TKeyProperty>>]
+  ? HasSanitizeParams<any, NotObjectCase<TOut, TKeyProperty>>
+  : [T] extends [HasInitPreContext<any, ObjectValueToAny<TOut, TKeyProperty>>]
+  ? HasInitPreContext<any, NotObjectCase<TOut, TKeyProperty>>
+  : never;
+
```

Comment:
it's possible some of the problems you were trying to fix would be solved by just tightening the type constraint instead - i.e. T extends HasPreAuthorize<object, any>
- [ ] explore tightening type constraint
...the only thing you need to think/worry about/handle is to figure out if that'll break undefined cases, and if you need a slight workaround for that (e.g. object | undefined)